pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -412,6 +418,24 @@ func (vm *volumeManager) getUnmountedVolumes(podName types.UniquePodName, expect
 	return filterUnmountedVolumes(mountedVolumes, expectedVolumes)
 }
 
+// recordVolumeMetrics runs a metrics-gathering function in a loop.
+func (vm *volumeManager) recordVolumeMetrics(stopCh <-chan struct{}) {
+	metricsLoop := func() {
+		metrics.ResetVolumeManagerVolumesMetric()
```

Comment:
Why are we resetting the metrics?

Reply:
Since `RecordVolumeManagerVolumesMetric` increments the counter, and this function is running in a loop every 10 seconds, the number of volumes in ASW and DSW would only grow without the reset.

Another option would be to store these numbers in a map, and then iterator over it and `Set()` and counter instead of `Inc()`ing it.